src/components/message/message.scss

```diff
@@ -0,0 +1,55 @@
+@use "../../design-tokens/colors.yml";
+@use "../../mixins/focus";
+@use "../../design-tokens/sizes.yml";
+@use "../../mixins/ms";
+
+/**
+ * 1. Push message and close button as far apart as possible.
+ * 3. When text breaks onto multiple lines our default line-height visually
+ *    competes with this element's padding, so we condense it a little.
+ */
+
+.c-message {
+  align-items: stretch;
+  background-color: colors.$gray-lighter;
+  color: colors.$text-dark;
+  display: flex; /* 1 */
+  justify-content: space-between; /*  */
+  line-height: ms.step(1, 1); /* 3 */
+}
+
+.c-message__content {
+  padding: ms.step(1);
+}
+
+/**
+ * 1. Button background-color should inherit the same color as its parent.
+ * 3. Close icon should be the same color as the parent text color.
+ * 4. Easy way to collapse extra whitespace between button edge and icon.
+ * 5. Make icon a little larger than the parent text size.
+ * 6. Put some space between message and close button.
+ */
+
+.c-message__close {
+  align-items: center;
+  background-color: inherit; /* 1 */
+  border-radius: sizes.$radius-medium;
+  border-width: 0;
+  color: inherit; /* 4 */
+  cursor: pointer;
+  display: flex;
+  font-size: ms.step(1); /* 5 */
+  margin-left: ms.step(1); /* 6 */
+  padding: 0 ms.step(1);
+
+  @include focus.focus-visible {
+    box-shadow: 0 0 0 sizes.$edge-large colors.$primary-brand-lighter;
+  }
+}
+
+.c-message__close:hover {
+  .c-icon {
```

Comment:
It's not really in the spirit of our guidelines to directly style a separate component from another. Some alternatives to consider:

1. Add a new class to the `c-icon` instance, for example `c-message__close-icon`. ([More info](https://www.smashingmagazine.com/2016/06/battling-bem-extended-edition-common-problems-and-how-to-avoid-them/#4-cross-component-components))
1. Apply the transform to `c-message__close` directly.